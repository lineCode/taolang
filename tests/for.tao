function main() {
    for1();
    for2();
    for3();
    for4();
}

function for1() {
    for {
        println("for1");
        break;
    }
}

function for2() {
    let n = 3;
    for n >= 0 {
        println(n);
        n = n - 1;
    }
}

function for3() {
    for let n = 3; n >= 0; n = n - 1 {
        println(n);
    }
}

function for4() {
    println("for4");
    for let i = 0; i < 3; i++ {
        let a = i;
        println(a);
    }
}
