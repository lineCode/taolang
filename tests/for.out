for1
3
2
1
0
3
2
1
0
for4
0
1
2
